src/constructors.jl

```diff
@@ -103,3 +103,24 @@ end
 function Base.call{T, N}(::Type{NullableArray{T, N}})
     NullableArray(T, ntuple(i->0, N))
 end
+
+#----- Conversion from Array of Nullables #1 -----------------------------------#
+# Take an Array of Nullable objects and convert it into a NullableArray
+function Base.convert{S,T,N}(::Type{NullableArray{S,N}},in::Array{Nullable{T},N})
+   out = NullableArray{S,N}(Array(S,size(in)),Array(Bool,size(in)))
+   for i = 1:length(in)
+       (out.isnull[i] = in[i].isnull) ? nothing : out.values[i] = in[i].value
+   end
+   out
+end
+
+#----- Conversion from Array of Nullables #2 -----------------------------------#
+# Take an Array of Nullable objects and convert it into a NullableArray
+# A slightly more convenient syntax to match the type of the input
+Base.convert{S,T,N}(::Type{NullableArray{S}},in::Array{Nullable{T},N}) = Base.convert(NullableArray{S,N},in)
```

Comment:
That's in constructor #8... some earlier constructors take precedence over this one if I just define `convert`.